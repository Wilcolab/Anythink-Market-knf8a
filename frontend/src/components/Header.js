import React from "react";
import { Link } from "react-router-dom";
import logo from "../imgs/topbar_logo.png";

const LoggedOutView = () => {
  return (
    <ul className="navbar-nav ml-auto">
      <li className="nav-item">
        <Link to="/login" className="nav-link">
          Sign in
        </Link>
      </li>

      <li className="nav-item">
        <Link to="/register" className="nav-link">
          Sign up
        </Link>
      </li>
    </ul>
  );
};

const LoggedInView = (props) => {
  return (
    <ul className="navbar-nav ml-auto">
      <li className="nav-item">
        <Link to="/editor" className="nav-link">
          <i className="ion-compose"></i>&nbsp;New Item
        </Link>
      </li>

      <li className="nav-item">
        <Link to="/settings" className="nav-link">
          <i className="ion-gear-a"></i>&nbsp;Settings
        </Link>
      </li>

      <li className="nav-item">
        <Link to={`/@${props.currentUser.username}`} className="nav-link">
          <img
            src={props.currentUser.image}
            className="user-pic pr-1"
            alt={props.currentUser.username}
          />
          {props.currentUser.username}
        </Link>
      </li>
    </ul>
  );
};

class Header extends React.Component {
  render() {
    return (
      <nav
        className="navbar navbar-expand-md navbar-dark"
        style={{ padding: "0.5rem 2rem" }}
      >
        <Link to="/" className="navbar-brand">
          <img alt="logo" src={logo} />
        </Link>

        {this.props.currentUser ? (
          <LoggedInView currentUser={this.props.currentUser} />
        ) : (
          <LoggedOutView currentUser={this.props.currentUser} />
        )}
      </nav>
    );
  }
}

export default Header;
