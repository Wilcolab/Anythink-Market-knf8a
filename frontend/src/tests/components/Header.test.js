import { create } from "react-test-renderer";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import Header from "../../components/Header";

describe("Header component", () => {
  it("Snapshot testing with no user", () => {
    const component = create(
      <Router>
        <Header />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Snapshot testing with user", () => {
    const component = create(
      <Router>
        <Header currentUser={{ username: "user name", image: "image.png" }} />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Check link to main page", () => {
    const header = mount(
      <Router>
        <Header />
      </Router>
    );
    expect(header.find("Link").first().prop("to")).toEqual("/");
  });

  it("Render register button when there's no user", () => {
    const header = mount(
      <Router>
        <Header />
      </Router>
    );
    expect(header.find("li > Link").first().text()).toEqual("Sign in");
  });

  it("Render user name when there's a user", () => {
    const user = { username: "user name", image: "image.png" };
    const header = mount(
      <Router>
        <Header currentUser={user} />
      </Router>
    );
    expect(header.find("li > Link").last().text()).toEqual(user.username);
  });
});
