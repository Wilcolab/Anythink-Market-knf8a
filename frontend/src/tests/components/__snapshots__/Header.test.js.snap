// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Snapshot testing with no user 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark"
  style={
    Object {
      "padding": "0.5rem 2rem",
    }
  }
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      src="topbar_logo.png"
    />
  </a>
  <ul
    className="navbar-nav ml-auto"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/register"
        onClick={[Function]}
      >
        Sign up
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Header component Snapshot testing with user 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark"
  style={
    Object {
      "padding": "0.5rem 2rem",
    }
  }
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      src="topbar_logo.png"
    />
  </a>
  <ul
    className="navbar-nav ml-auto"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/editor"
        onClick={[Function]}
      >
        <i
          className="ion-compose"
        />
         New Item
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/settings"
        onClick={[Function]}
      >
        <i
          className="ion-gear-a"
        />
         Settings
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/@user name"
        onClick={[Function]}
      >
        <img
          alt="user name"
          className="user-pic pr-1"
          src="image.png"
        />
        user name
      </a>
    </li>
  </ul>
</nav>
`;
